### DISCUSSION TITLE
Responsive charts outside of Observable

### DISCUSSION THREAD

**Discussion by dgwyer** [observablehq/plot#974] [Q&A] [answered]
From the docs:

> The default width is 640. On Observable, the width can be set to the [standard width](https://github.com/observablehq/stdlib/blob/main/README.md#width) to make responsive plots.

I need to be able to do the same for my own plots outside of the Observable website. Is there a way to do that right now?

**Comment by Fil** [comment]
There are as many ways as there are frameworks.


In vanilla JavaScript, you might want to listen to the resize event and refresh the plot:

```js
doChart();
window.addEventListener("resize", doChart);
function doChart() {
    document.querySelector("#chart").innerHTML = "";
    document
      .querySelector("#chart")
      .appendChild(
        Plot.plot({ marks: [Plot.frame()], width: window.innerWidth })
      );
  });
```

**Comment by dgwyer** [comment, marked as answer]
Thanks for posting that.

Would it also be an idea to have an option that set the SVG width to 100% the size of the container rather than a specific width. That would be super useful.

**Reply by Fil** [reply]
Plot returns an HTML or SVG node, but has no notion of where it will be contained. You can however give it a style (as the top-level **style** option), or add anything you want to it in vanilla JavaScript (a className, for example) before you append it to the container.

**Reply by dgwyer** [reply]
I meant that at the moment a fixed width/height is always(?) applied to the SVG node returned by plot. But in some cases I need to not have this explicitly set so it can take up the full width of its container. I don't suppose that's an option at the moment?

**Reply by Fil** [reply]
~~You can remove it with setAttribute:~~ 

```
  const a = Plot.plot({ marks: [Plot.frame()] });
  a.setAttribute("width", null);
  return a;
```

EDIT: better version below.

**Reply by dgwyer** [reply]
While this works, it does result in a console error for me: `Error: <svg> attribute width: Expected length, "null".`.

I think we need an additional option to be able to remove the `width` SVG attribute altogether. And do the same for the `height` attribute too, as a separate option, in case we need a fixed height, but responsive width.

![Frame 7](https://user-images.githubusercontent.com/1482075/177626060-49d48209-2bda-4f98-b83a-01b0797e1f31.png)

**Reply by Fil** [reply]
sorry, I should have double-checked. Here's a better version:
```
  const a = Plot.plot({ marks: [Plot.frame()] });
  a.removeAttribute("width");
  a.removeAttribute("height");
  return a;
```

**Reply by dgwyer** [reply]
No problem, and thanks for the update. That should work nicely.
